new-components/reporters/slack/internal/reporter/reporter.go

```diff
@@ -0,0 +1,146 @@
+package reporter
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/go-errors/errors"
+	"github.com/smithy-security/pkg/env"
+
+	"github.com/smithy-security/smithy/sdk/component"
+	vf "github.com/smithy-security/smithy/sdk/component/vulnerability-finding"
+	ocsffindinginfo "github.com/smithy-security/smithy/sdk/gen/ocsf_ext/finding_info/v1"
+)
+
+type (
+	Conf struct {
+		SlackWebhook string
+	}
+
+	slackLogger struct {
+		Conf *Conf
+	}
+
+	SlackRequestBody struct {
+		Text string `json:"text"`
+	}
+)
+
+// NewSlackLogger returns a new slack logger.
+func NewSlackLogger(c *Conf) (*slackLogger, error) {
+	if c == nil {
+		return nil, errors.Errorf("configuration is nil")
+	}
+	return &slackLogger{
+		Conf: c,
+	}, nil
+}
+
+// NewConf returns a new configuration build from environment lookup.
+func NewConf(envLoader env.Loader) (*Conf, error) {
+	var envOpts = make([]env.ParseOption, 0)
+	if envLoader != nil {
+		envOpts = append(envOpts, env.WithLoader(envLoader))
+	}
+
+	webhook, err := env.GetOrDefault(
+		"SLACK_WEBHOOK",
+		"",
+		append(envOpts, env.WithDefaultOnError(false))...,
+	)
+	if err != nil {
+		return nil, errors.Errorf("could not get env variable for SLACK_WEBHOOK: %w", err)
+	}
+
+	return &Conf{
+		SlackWebhook: webhook,
+	}, nil
+}
+
+// Report logs the findings summary in slack.
+func (s slackLogger) Report(
+	ctx context.Context,
+	findings []*vf.VulnerabilityFinding,
+) error {
+	logger := component.
+		LoggerFromContext(ctx).
+		With(slog.Int("num_findings", len(findings)))
+
+	if len(findings) > 0 {
+		scanID := findings[0].Finding.FindingInfo.Uid
+		numResults := s.countFindings(findings)
+		startTime := findings[0].Finding.StartTime
+		asTime := time.Unix(*startTime, 0).UTC()
+		toDatetime := asTime.Format(time.RFC3339)
+		newFindings := s.countNewFindings(findings)
+		msg := fmt.Sprintf("Smithy scan %s, started at %s, completed with %d findings, out of which %d new", scanID, toDatetime, numResults, newFindings)
+		return s.push(ctx, msg, s.Conf.SlackWebhook)
+	}
+	logger.Error("Received 0 scans, this is likely an error, skipping sending empty message")
+
+	return nil
+}
+
+func (s slackLogger) push(ctx context.Context, b string, webhook string) error {
+
+	var err error
+	body, err := json.Marshal(SlackRequestBody{Text: b})
+	if err != nil {
+		return err
+	}
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, webhook, bytes.NewBuffer(body))
```

Comment:
Can we leverage a context with timeout here to make sure that we're not stuck forever? 10 seconds should do